check.sh

```diff
@@ -20,27 +20,47 @@ if [[ -n "$noduplicate" ]];then
   exit 1
 fi
 
-echo "CHECK that deleted translator are added to deleted.txt and number is increased"
-deleted=$(git diff --cached --name-only --diff-filter=D -- '*.js*')
-if [[ -n "$deleted" ]];then
-  # number increased
-  oldrevision=$(git show HEAD:deleted.txt | awk 'NR<2 { print $1 }')
-  newrevision=$(awk 'NR<2 { print $1 }' deleted.txt)
-  if [ $newrevision -ne $(($oldrevision + 1)) ];then
-    echo "Error: translators are deleted but the number in deleted.txt is not increased by 1."
-    echo "$newrevision (newrevision) =/= $oldrevision (oldrevision) + 1"
-    exit 1
-  fi
-  # all deleted translator ids must be now listed in deleted.txt
-  listed=$(git diff --cached --diff-filter=D -- '*.js*' | grep '"translatorID"' | awk -F: '{print $2}' | sed 's/[", ]//g' | xargs -i grep -c {} deleted.txt | awk '$1!="1" {print "Error"}')
-  if [[ -n "$listed" ]];then
-    echo "Error: translators are deleted but not listed in deleted.txt."
-    exit 1
+
+#list all commits where files are deleted
+commits=$(git --no-pager log --diff-filter=D --pretty="%H")
+first=true
+for c in $commits; do
+  echo "$c"
+  #check that a JavaScript file (i.e. a translator) is deleted
+  jsdeleted=$(git show --name-status $c | grep ^D.*\.js | sed 's/^D\s*//')
+  if [[ -n "$jsdeleted" ]];then
+    if [ "$first" = true ];then
+      before=$(git log -n 2 $c --pretty=format:"%H" | tail -1)
+      # number increased
+      oldrevision=$(git show "$before":deleted.txt | awk 'NR<2 { print $1 }')
+      newrevision=$(awk 'NR<2 { print $1 }' deleted.txt)
+      if [ $newrevision -ne $(($oldrevision + 1)) ];then
+        echo "Error: translators are deleted but the number in deleted.txt is not increased by 1 (compared to commit $before)."
+        echo "$newrevision (newrevision) =/= $oldrevision (oldrevision) + 1"
+        exit 1
+      fi
+      first=false
+    fi
+    
+    # all deleted translator ids must be listed in deleted.txt
+    #echo $jsdeleted
+    #git show "$c" --diff-filter=D | grep '"translatorID"'
+    delids=$(git show "$c" --diff-filter=D | grep '"translatorID"' | awk -F: '{print $2}' | sed 's/[", ]//g')
+    listed=$(echo $delids | xargs -i grep -c {} deleted.txt | awk '$1!="1" {print "Error"}')
+    rename=$(cat *.js | grep -c $delids | awk '$1!="1" {print "Error"}')
+    if [[ (-n "$listed") && (-n "$rename") ]];then
+      echo "Error: translators are deleted but not listed in deleted.txt."
+      echo "Commit: $c"
+      echo "Deleted translator: $jsdeleted ; with id $delids"
+      echo $delids | xargs -i grep -c {} *.js
+      exit 1
+    fi
+    
```

Comment:
Uh, I missed that comment completely. The reason I tried here to go through the whole git history is that I can imagine a lot of different scenarios:
 * I make a PR with a commit that deletes a translator but Travis check fails. For fixing this I will make another commit where deleted.txt is changed. ==> It is not enough to check the changes of individual commits, but we have to check the changes of the whole PR.
 * A maintainer is pushing directly to master branch without PRs ==> We cannot compare the changes with master branch, in order to make Travis check useful on master branch itself. (Also any build status because master would always pass Travis check, would be very misleading.)
 * It is possible that deleting a translator is just one change of a commit together with other changes (e.g. https://github.com/zotero/translators/commit/e102b02ae99d8111e44d981d19518ad1f4ded69b) and we can delete several translators in the same commit.

What do you think?

> We don't need to go through the whole history on every commit.

Well, that is in principle true, but I tried to outline that it might be hard to know how much of the git history we have to check in each possible use case. And on the other hand, it does not hurt us, if we go trough the whole history every time.